src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,97 @@
+package duke.utils;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.Date;
+import duke.DukeException;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteTaskCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.MarkTaskCommand;
+import duke.command.PrintListCommand;
+import duke.command.ToDoCommand;
+import duke.command.UnmarkTaskCommand;
+
+
+/**
+ * Parses tbe input received by the AIlfred bot.
+ * @author Jason
+ */
+public class Parser {
+
+    /**
+     * Handles the parsing of dates to create Deadlines or Events.
+     * @param input String input of date.
+     * @return Date object used to construct Deadlines or Events.
+     * @throws DukeException Exception if date format is wrong.
+     */
+    public static Date parseDate(String input) throws DukeException {
+        try {
+            DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            LocalDate parsed = LocalDate.parse(input, dtf);
+
+            return new Date(parsed);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("☹ OOPS!!! Please follow the Date and Time Format: yyyy-MM-dd [2000-01-01]");
+        }
+
+    }
+
+    /**
+     * Handles the parsing of dates from save file data for Deadlines or Events.
+     * @param input String input of date.
+     * @return Date object used to construct Deadlines or Events.
+     * @throws DukeException Exception if date format is wrong.
+     */
+    public static Date parseDateSave(String input) throws DukeException {

```

Comment:
I like your method names. Written in verb form and camelCase, it is easy to understand!